diff --git a/lua/glance/lsp.lua b/lua/glance/lsp.lua
--- a/lua/glance/lsp.lua
+++ b/lua/glance/lsp.lua
@@ -29,13 +29,13 @@ local function create_handler(method)
         if result == nil or vim.tbl_isempty(result) then
           client_request_ids[ctx.client_id] = nil
         else
+          local islist = vim.fn.has('nvim-0.10.0') == 1 and vim.islist
+            or vim.tbl_islist
+
           cancel_all_requests()
-          result = (
-            vim.fn.has('nvim-0.10.0') == 1 and vim.islist(result)
-            or vim.tbl_islist(result)
-          )
-              and result
-            or { result }
+          result = islist(result)
+              and result
+              or { result }
 
           return cb(result, ctx)
         end
